### PR TITLE
User management

### LINKED ISSUE #33: Implement Identification Mechanism
Develop a secure identification system with login, logout, registration, and refresh token functionalities. Use JWT for authentication and authorization.

### LINKED ISSUE #59: add 3 requests
GetSuggestionByEventId - We want to display the last suggestion when the user enter to the website in the first time (if there are events from the last date)

CreateDailySharing (with mood) - the client send the dailySharing. the server add the mood (using the gemini), date, dailySharing, create an item of event and return the new event.

UpdateSuggestion (with rank, executionDate, etc) - when the user click on differenct buttons in the sharing page (like, didnt like, etc), we want to update the rank and the executiondate of the suggestion, and return the updatedSuggestion

### PR DESCRIPTION
**Pull request by amitkeinan9**
Added login/logout/refresh and register routes

Closes #33

### CODE REVIEW

**Comment by amitkeinan9** [comment]
Accidentally pushed here the change for #59.

I changes the API for generating the suggestion as well as the flow. @yaelili70 take a look at the last commit for the routes you asked for and how to generate suggestions, get them and edit them.

File: Backend/prisma/schema.prisma [hunk 1]
```diff
@@ -25,10 +24,17 @@ model suggestions {
   content        String
```
Comment (yaelili70) [inline Backend/prisma/schema.prisma:32]: remove title for suggestion, there is only the content

File: Backend/src/repository/suggestions.repository.ts [hunk 2]
```diff
@@ -1,34 +1,96 @@
...
+
+export const saveNewSuggestion = async (
+  prisma: PrismaClient,
+  suggestion: string,
+  eventId: number
+) =>
+  prisma.suggestions.create({
+    data: {
+      content: suggestion,
+      title: "", // TODO: Do we need the title?
```
Comment (yaelili70) [inline Backend/src/repository/suggestions.repository.ts:34]: no

File: Backend/prisma/schema.prisma (hunk 1 above)
Comment (amitkeinan9) [reply Backend/prisma/schema.prisma:32]: I might add it later so we will leave it for now

File: Backend/src/routers/auth/authRouter.spec.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+import { AxiosRequestConfig } from "axios";
+import { send } from "process";
+
+const { PORT } = Configuration.getInstance();
+
+let accessToken: string, refreshToken: string;
+
+const registerUser = async (body: Record<string, unknown>) => {
+  const res = await testAxios.post(
+    `http://localhost:${PORT}/auth/register`,
```
Comment (yaelili70) [inline Backend/src/routers/auth/authRouter.spec.ts:15]: we need env files to save the url?

File: Backend/vitest-api-setup.ts [hunk 4]
```diff
@@ -1,15 +1,28 @@
...
   configDotenv({
     path: "./.env.test",
   });
 
-  killServer = initServer();
+  killServer = await initServer();
+
+  const res = await testAxios.post(`http://localhost:${PORT}/auth/login`, {
+    email: "john.doe@example.com",
+    password: "password123",
```
Comment (yaelili70) [inline Backend/vitest-api-setup.ts:22]: it is not secure...

**Review by yaelili70 — APPROVED** [review]
Good job Amit!

**Pull request merged**